.travis-ci.sh

```diff
@@ -53,26 +52,33 @@ critical errors that must be understood and fixed before your pull
 request can be merged.
 ------------------------------------------------------------------------
 EOF
-    mkdir -p $PREFIX
+  case $XARCH in
+  x64)
     ./configure --prefix $PREFIX -with-debug-runtime \
       -with-instrumented-runtime $CONFIG_ARG
-    export PATH=$PREFIX/bin:$PATH
-    $MAKE world.opt
-    $MAKE ocamlnat
-    (cd testsuite && $MAKE all)
-    (cd testsuite && $MAKE USE_RUNTIME="d" all)
-    $MAKE install
-    # check_all_arches checks tries to compile all backends in place,
-    # we need to redo (small parts of) world.opt afterwards
-    $MAKE check_all_arches
-    $MAKE world.opt
-    $MAKE manual-pregen
+    ;;
+  i386)
+    ./configure --prefix $PREFIX -with-debug-runtime \
+      -with-instrumented-runtime $CONFIG_ARG \
+      -host i686-pc-linux-gnu
     ;;
   *)
     echo unknown arch
     exit 1
     ;;
   esac
+
+  export PATH=$PREFIX/bin:$PATH
+  $MAKE world.opt
+  $MAKE ocamlnat
+  (cd testsuite && $MAKE all)
+  [ $XARCH =  "i386" ] ||  (cd testsuite && $MAKE USE_RUNTIME="d" all)
+  $MAKE install
+  # check_all_arches checks tries to compile all backends in place,
+  # we need to redo (small parts of) world.opt afterwards
```

Comment:
Why do we need to redo `world.opt`?  Since the manual won't build?  If so can we just put the manual build before `check_all_arches` to save time?

Reply:
I don't know about this - need to look at the git blame to see who originated the comment! I think it's to do with running the testsuite, rather than building the manual, though

Reply:
The testsuite has already been run by that time, no?

Reply:
Oh ah (the curse of reviewing on a phone...). Actually, this was necessary before when ocamlbuild and camlp4 were subsequently compiled, but that's no longer an issue, of course.